sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc

```diff
@@ -188,15 +199,30 @@ constant unsigned 32-bit integer. Behavior is undefined when 'Column Count' is
 0 or <<OpConstantNull,*OpConstantNull*>>. +
  +
 'Layout' indicates how the values are arranged internally in the matrix type. 
-It must be the result of a constant instruction. +
+Must be an '<id>' of a 32-bit integer scalar. Its value is expected to be one of
+the values in the table 3.XX, Matrix Layout. If validation rules or the client API
+require a constant '<id>', it is invalid for it to not be one of these values.
+If non-constant '<id>' are allowed, behavior is undefined if '<id>' is not one of
+these values. +
+ +
+'Scope' is memory scope for operations on the matrix.
+Must be an '<id>' of a 32-bit integer scalar. Its value is expected to be one
+either _Workgroup_ or _Subgroup_ from the table 3.27. Scope <id>. If validation
+rules or the client API require a constant '<id>', it is invalid for it to not
+be one of these values. If non-constant '<id>' are allowed, behavior is
+undefined if '<id>' is not one of these values. +
+ +
+'Use' parameter shows where in a math operation the matrix is used.
+Must be an '<id>' of a 32-bit integer scalar. Its value is expected to be one of
+the values in the table 3.XX, Matrix Use. If validation rules or the client API
+require a constant '<id>', it is invalid for it to not be one of these values.
+If non-constant '<id>' are allowed, behavior is undefined if '<id>' is not one of
+these values. +
```

Comment:
The plan is to remove Layout from the matrix type (in rev 7 I guess). We are keeping it now to not break our own implementation.

>> Is the "Layout" ever observable, or is this more of a hint that could improve performance of loads and stores in some cases?

Yes, it's a hint for loads and stores, just like 'ColumnMajor' parameter in SPV_NV_cooperative_matrix . 

>> In most cases, won't the "layout" be implied by the "use"?

Correct, in our implementation we even started to rewrite layout depending on use during SPIR-V consumption.

Reply:
I think we should explicitly describe the layout on the type as a hint then, both to motivate why it may eventually go away, and to clearly differentiate from the layout on the load and store instructions that are not a hint and are required.